bouncer/src/repo/manipulator/modelconvertor/import/repo_model_import_3drepo.cpp

```diff
@@ -132,25 +137,76 @@ repo::core::model::MaterialNode *RepoModelImport::parseMaterial(const boost::pro
 	materials.insert(materialNode);
 	matNodeList.push_back(materialNode);
 
-	return materialNode;
+	if(textureId >= 0)
+	{
+		textureIdToParents[textureId].push_back(materialNode->getSharedID());
+	}
+	return;
```

Comment:
this is not needed?

Reply:
This line is required to match up a texture to its parent material nodes. It's used later down the line when a texture is parsed https://github.com/3drepo/3drepobouncer/blob/6f5db153802ccb66afd39eb354756a7eabcfbdcb/bouncer/src/repo/manipulator/modelconvertor/import/repo_model_import_3drepo.cpp#L188

Reply:
@har00n-haider  I don't mean adding into `textureIdToParents`

I mean line 144, the return just before the end of the function

Reply:
oops, I dunno why my eyes were drawn to that other line....

I'll remove the return!